Mathlib/Data/List/Basic.lean

```diff
@@ -1756,6 +1756,16 @@ theorem bind_congr {l : List α} {f g : α → List β} (h : ∀ x ∈ l, f x =
   (congr_arg List.join <| map_congr h : _)
 #align list.bind_congr List.bind_congr
 
+theorem bind_option_toList_of_eq_some {f : β → Option α} {g : β → α} {bs : List β}
+    (h : ∀ x ∈ bs, f x = some (g x)) : bs.bind (fun i => (f i).toList) = bs.map g := by
+  have : bs.bind (fun i => (f i).toList) = bs.bind (List.ret ∘ g) :=
+    List.bind_congr (by simp; intro m hm; simp[h _ hm]; rfl)
+  rw [this, List.bind_ret_eq_map]
+
+theorem subset_bind_of_mem {a : α} {as : List α} (h : a ∈ as) (f : α → List α) :
+    f a ⊆ as.bind f := by
```

Comment:
This could be strengthened to `List.infix`

Reply:
add `infix_bind_of_mem`